flocker/apiclient/_client.py

```diff
@@ -751,6 +772,26 @@ def version(self):
             b"GET", b"/version", None, {OK}
         )
 
+    @staticmethod
+    def _parse_volumes(volume_list):
+        """
+        Parse a list decoded from JSON with volume configuration.
+
+        :param Optional[Sequence[Mapping[unicode, unicode]]] volume_list: List
+            describing a JSON list of volume objects.
+        :return Optional[Sequence[MountedDataset]]: List of mounted datasets,
+            or None if no volumes.
+        """
+        if volume_list:
+            return [
+                MountedDataset(
+                    dataset_id=UUID(volume[u'dataset_id']),
+                    mountpoint=volume[u'mountpoint'],
+                ) for volume in volume_list
+            ]
```

Comment:
Generator expressions are iterators, but the return value must be a restartable iterable.